### ISSUE TITLE
[BUG] azure-core-test need to support Jackson 2.15.0

### ISSUE THREAD

**Issue by Netyyyy** [Azure/azure-sdk-for-java#35602]
**Describe the bug**
refer to https://github.com/FasterXML/jackson-core/issues/1001, we have the same error [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=2870875&view=logs&j=c9746e28-7f0e-5889-d24a-696559b4d72e&t=983c3bb1-387f-5b37-4748-b478b041554d&l=20042) after using jackson 2.15

***Exception or Stack Trace***
```
2023-06-25T07:11:31.8166142Z [ERROR] Errors: 
2023-06-25T07:11:31.8168135Z [ERROR]   BlockBlobInputOutputStreamTest>StorageSpec.setup:45 » UncheckedIO com.fasterxml.jackson.databind.JsonMappingException: String length (5047468) exceeds the maximum length (5000000) (through reference chain: com.azure.core.test.models.RecordedData["networkCallRecords"]->java.util.LinkedList[2]->com.azure.core.test.models.NetworkCallRecord["Response"])
2023-06-25T07:11:31.8169923Z [ERROR]   AppendBlobAPITest>StorageSpec.setup:45 » UncheckedIO com.fasterxml.jackson.databind.JsonMappingException: String length (5041918) exceeds the maximum length (5000000) (through reference chain: com.azure.core.test.models.RecordedData["networkCallRecords"]->java.util.LinkedList[3]->com.azure.core.test.models.NetworkCallRecord["Response"])
```

**Expected behavior**
Fix the error on Jackson 2.15

**Comment by Netyyyy** [comment]
@alzimmermsft please help take a look, thanks!

**Comment by alzimmermsft** [comment]
Fixed by #35659
